### PR TITLE
feat(desktop): gate CDP remote debugging behind an explicit evaluator

### PR DESCRIPTION
**Pull request by Sora-bluesky**
## Summary

The desktop app now decides for itself whether to compose CDP remote debugging arguments, instead of
taking that decision from an unvalidated environment value. A new pure evaluator takes two signals —
a test profile and a port — and returns an explicit gate decision. Both signals must be present and
valid before debugging is enabled, and the profile is matched against a single accepted token rather
than any truthy value.

This governs the arguments the application itself supplies. It does not close the separate
`WEBVIEW2_ADDITIONAL_BROWSER_ARGUMENTS` channel, which the WebView2 runtime reads directly and which
the bakeoff launcher and the default public-smoke mode still use on purpose. A parent that sets a
debugging switch there can still reach the runtime without either gate variable. Neutralising that
channel would require moving both callers onto the gate variables first, which is not attempted here;
this change neither widens nor narrows it.

Port handling rejects the whole class of inputs that used to slip through: empty strings, whitespace
padding, non-numeric text, zero, values outside the u16 range, and anything below a policy floor of
1024. A stray port with no profile set is ignored rather than honoured. When the gate is off, which is
the default, no debugging arguments are composed at all.

The gate variables are also scrubbed from the environment of child processes the app spawns, so an
enabled debugging session cannot silently propagate into unrelated processes started later.

Public-release smoke gains a matching observation path: it can now watch the app-side injection route
and assert on the page URL actually loaded, rather than inferring from launch side effects.

## Why

The application's own debugging port was previously taken from an environment value that no component
owned. Nothing validated it, and nothing prevented it from being inherited by processes spawned
later. That is a wide surface for a shipped desktop application, and it is not observable from the
outside: a build with debugging enabled looks the same as one without.

Making the decision explicit and pure means it can be exhaustively tested, and moves the default to
off rather than to whatever the environment says.

## Verification

The Rust decision-table units pass: 223 passed, 0 failed, 0 ignored, exit code 0. They cover profile
absence, stray ports without a profile, empty and whitespace-padded and non-numeric ports, port zero,
out-of-range values, the policy floor, the accepted boundary values, and unrecognized profile tokens.

The Pester registration gate passes: exit code 0, status covered, 13 shards over 18 files.

Of the three owning Pester suites, 10 tests pass and 1 fails locally. The failure names a required
environment variable that points at a built application binary. VersionSurface timed out on this
machine and produced no counts, so it is neither a pass nor a failure here.

## Scope and rollback

The behavioural change is the gate evaluator, its wiring into app startup, the child-environment
scrub, and the public smoke observation path, together with their tests. The rest of the diff is
supporting material: a threat-model entry recording the new surface in both the model and its audit
table, the corresponding update to the threat-model contract test that counts those entries,
commit-whitelist and ignore-list registrations for the newly tracked files, one Tauri configuration
flag, and additions to the existing VersionSurface suite.

The new candidate workflow is dispatch-only. It has no push or pull-request trigger and will not run
by itself; it is triggered by hand against a selected candidate commit, which it checks out and
builds, so treating it as an automatic safety net would be a mistake. Rollback is a single-branch
revert.

### CODE REVIEW

File: winsmux-app/src-tauri/src/lib.rs [hunk 1]
```diff
@@ -1805,6 +1820,19 @@ impl PtyCommandTransport for TauriPtyTransport {
...
 pub fn run() {
+    let remote_debug_browser_args: Option<String> = match remote_debug_gate::resolve_from_env() {
+        remote_debug_gate::RemoteDebugGate::Rejected(reason) => {
+            eprintln!(
+                "winsmux-desktop: remote debug gate rejected: {}",
+                reason.diagnostic_token()
+            );
+            std::process::exit(2);
+        }
+        remote_debug_gate::RemoteDebugGate::Disabled => None,
```
[the remark this answers is not in the document]
Comment (Sora-bluesky) [reply winsmux-app/src-tauri/src/lib.rs:1831]: Confirmed against the source. That variable is set at scripts/start-cli-bakeoff-desktop.ps1:916 and scripts/test-public-release.ps1:2623, and appears in no Rust source, so the runtime can still receive a debugging switch without either gate variable.

This is not fixed here. Both callers use that channel on purpose today, and the default public-smoke mode PortAndMarker still puts the port there, so removing the switch would break them until they move onto the gate variables.

The pull-request body has been corrected: it no longer claims the app decides instead of inheriting from the environment, and it now states that this channel remains open.

[the remark this answers is not in the document]
Comment (Sora-bluesky) [reply winsmux-app/src-tauri/src/lib.rs:1831]: Filed as #1250 with the evidence, the two callers that keep the channel in use, and the migration order this needs. Resolving this thread against that issue rather than a code change here.
